src/imp/atomic128/x86_64.rs

```diff
@@ -144,25 +144,126 @@ unsafe fn cmpxchg16b(
     }
 }
 
+// VMOVDQA is atomic on on Intel CPU with AVX.
+// See https://gcc.gnu.org/bugzilla//show_bug.cgi?id=104688 for details.
+//
+// Do not use vector registers on targets such as x86_64-unknown-none unless SSE is explicitly enabled.
+// https://doc.rust-lang.org/nightly/rustc/platform-support/x86_64-unknown-none.html
+#[cfg(target_feature = "sse")]
+#[target_feature(enable = "avx")]
+#[inline]
+unsafe fn _atomic_load_vmovdqa(src: *mut u128, _order: Ordering) -> u128 {
+    debug_assert!(src as usize % 16 == 0);
+    // SAFETY: the caller must uphold the safety contract for `_atomic_load_vmovdqa`.
+    unsafe {
+        let out: core::arch::x86_64::__m128;
+        asm!(
+            "vmovdqa {out}, xmmword ptr [{src}]",
+            src = in(reg) src,
+            out = out(xmm_reg) out,
+        );
+        core::mem::transmute(out)
+    }
+}
+#[cfg(target_feature = "sse")]
+#[target_feature(enable = "avx")]
+#[inline]
+unsafe fn _atomic_store_vmovdqa(dst: *mut u128, val: u128, _order: Ordering) {
+    debug_assert!(dst as usize % 16 == 0);
+    // SAFETY: the caller must uphold the safety contract for `_atomic_store_vmovdqa`.
+    unsafe {
+        let val: core::arch::x86_64::__m128 = core::mem::transmute(val);
+        asm!(
+            "vmovdqa xmmword ptr [{dst}], {val}",
+            // TODO: GCC always uses mfence here, so the current implementation
+            // follows that behavior. However, probably only SeqCst store
+            // actually requires mfence.
+            "mfence",
```

Comment:
I'm not sure why GCC always uses mfence.

Reply:
Yeah I don't think this needs an `mfence` unless it's `SeqCst` as with regular atomic stores. GCC might have overlooked it? Probably worth filing an issue there.